krystex/src/main/java/com/flipkart/krystal/krystex/logicdecorators/observability/DefaultKryonExecutionReport.java

```diff
@@ -136,6 +148,35 @@ private Map<ImmutableMap<String, Object>, Object> convertResult(Results<Object>
                 e -> extractAndConvertInputs(e.getKey()), e -> convertValueOrError(e.getValue())));
   }
 
+  public static <T> String hashValues(T input) {
+    StringBuilder concatenatedValues = new StringBuilder();
+    if (input != null) {
+      concatenatedValues.append(input.toString());
+    }
+    return hashString(concatenatedValues.toString());
+  }
+
+  private static String hashString(String appendedInput) {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      byte[] encodedHash = digest.digest(appendedInput.getBytes(StandardCharsets.UTF_8));
+      StringBuilder hexString = new StringBuilder(2 * encodedHash.length);
+
+      for (byte b : encodedHash) {
+        String hex = Integer.toHexString(0xff & b);
+        if (hex.length() == 1) {
+          hexString.append('0');
+        }
+        hexString.append(hex);
+      }
+
+      return hexString.toString();
+    } catch (NoSuchAlgorithmException e) {
+      log.error("Error came while generating message digest instance.");
+      return "";
```

Comment:
Give detailed message as return value
`Error: could not hash inputs because of exception: ` etc

Reply:
The exception e must also be part of the error log